pills/08/generic-builder.txt

```diff
@@ -1,6 +1,6 @@
 set -e
 unset PATH
-for p in $buildInputs; do
+for p in $baseInputs; do
```

Comment:
I don't think `builder.sh` is expected to be used as is https://nixos.org/guides/nix-pills/generic-builders.html#idm140737320207792 :

```
Exercise: Complete the new builder.sh by adding $baseInputs in the for loop together with $buildInputs. As you noticed, we passed that new variable in the derivation. Instead of merging buildInputs with the base ones, we prefer to preserve buildInputs as seen by the caller, so we keep them separated. Just a matter of choice. 
```

Note that you need to iterate through both lists to get the desired result.